src/variety/cbblock.js

```diff
@@ -233,6 +361,7 @@ AnsCheck:{
 		var sides = this.board.blockgraph.getSideAreaInfo();
 		for(var i=0;i<sides.length;i++){
 			var area1 = sides[i][0], area2 = sides[i][1];
+			if(area1.dotcnt!==2 || area2.dotcnt!==2) {continue;}
```

Comment:
I'm a bit confused by the various `dotcnt`-related changes, but then I'll wait to understand the error code stacking before putting too much work in here. :)

Reply:
In cbblock, `dotcnt` is the number of outlined regions contained in a block, and is only set on nodes in AreaTileGraph. I need to use the `isDifferentShapeBlock` function on other node types.

It seems neater to me to only check if the shapes are different in `isDifferentShapeBlock`, and not check if the shapes should even be compared in the first place. That should be the caller's responsibility.